docs/source/en/using-diffusers/sdxl.md

```diff
@@ -0,0 +1,425 @@
+    text_encoder_2=pipe.text_encoder_2,
+    vae=pipe.vae,
+    torch_dtype=torch.float16,
+    use_safetensors=True,
+    variant="fp16",
+).to("cuda")
+```
+
+Generate an image from the base model, and set the model output to **latent** space:
+
+```py
+prompt = "Astronaut in a jungle, cold color palette, muted colors, detailed, 8k"
+
+image = base(prompt=prompt, output_type="latent" if use_refiner else "pil").images[0]
+```
+
+Pass the generated image to the refiner model:
+
+```py
+image = refiner(prompt=prompt, image=image[None, :]).images[0]
+```
+
+<div class="flex gap-4">
+  <div>
+    <img class="rounded-xl" src="https://huggingface.co/datasets/diffusers/docs-images/resolve/main/sd_xl/init_image.png" alt="generated image of an astronaut riding a green horse on Mars" />
+    <figcaption class="mt-2 text-center text-sm text-gray-500">base model</figcaption>
+  </div>
+  <div>
+    <img class="rounded-xl" src="https://huggingface.co/datasets/diffusers/docs-images/resolve/main/sd_xl/refined_image.png" alt="higher quality generated image of an astronaut riding a green horse on Mars" />
+    <figcaption class="mt-2 text-center text-sm text-gray-500">base model + refiner model</figcaption>
+  </div>
+</div>
+
+For inpainting, load the refiner model in the [`StableDiffusionXLInpaintPipeline`], remove the `denoising_end` and `denoising_start` parameters, and choose a smaller number of inference steps for the refiner.
+
+## Micro-conditioning
+
+SDXL adds two addition conditioning techniques, which are referred to as *micro-conditioning*. The model is conditioned on image size so at inference, you can set a desired resolution for the generated image. The second conditioning is based on cropping parameters, allowing you to control how the generated image is cropped.
+
+<Tip>
+
+Size and crop-conditioning parameters can be used together to generate high-resolution images that are centered on a subject. These micro-conditionings and negative micro-conditionings are available in the [`StableDiffusionXLPipeline`], [`StableDiffusionXLImg2ImgPipeline`], [`StableDiffusionXLInpaintPipeline`], and [`StableDiffusionXLControlNetPipeline`].
+
+</Tip>
+
+### Size conditioning
+
+Size conditioning takes advantage of what SDXL has learned about image features at different resolutions during training to generate higher quality images during inference. You can experiment with this by adjusting the [`original_size`](https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/stable_diffusion_xl#diffusers.StableDiffusionXLPipeline.__call__.original_size) and [`target_size`](https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/stable_diffusion_xl#diffusers.StableDiffusionXLPipeline.__call__.target_size) parameters. By default, both parameters are set to 1024 to generate better images. If your `original_size` and `target_size` don't match, then the image is either down or upsampled to match the `target_size`.
```

Comment:
This is not how it works, conditioning only carries information about the data used during training and no resizing is used. I'd suggest something like:

```suggestion
If the source images in the training dataset are too small, we can either ignore them, or resize them to match the desired size. Small images account for almost 40% of the total training data used to train SDXL, so the authors decided to upscale and use them for training. However, the model also receives "original size" conditioning, which is just the information about the resolutions of the images in the training batch. This way, the model learns that upscaling artifacts are not supposed to be present in high-resolution images. At inference time, we can use [`original_size`](https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/stable_diffusion_xl#diffusers.StableDiffusionXLPipeline.__call__.original_size) to indicate the original resolution. If we use the recommended value of `(1024, 1024)`, the model will produce images more similar to the 1024x1024 images in the dataset, which in general will be higher quality than lower-resolution ones. If you use a low resolution, such as `(256, 256)`, the model will still produce 1024x1024 images, but it will try to _mimic_ the appearance of low-res images in the dataset: simpler patterns and blurring are to be expected!

The [`target_size`](https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/stable_diffusion_xl#diffusers.StableDiffusionXLPipeline.__call__.target_size), on the other hand, was used when the model was fine-tuned to support various aspect ratios, so it's intended to be used in that context too. For example, if you use `(1024, 1024)`, you'll get images that resemble the composition of square images in the dataset. We recommend you set the same value for this conditioning parameter as the resolution you are trying to create, but you are free to experiment with other options!
```

Feel free to simplify if it's too long.